LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/WebSocketWriterRegistry.cs

```diff
@@ -84,6 +84,16 @@ public bool TryGetHandle(TKey key, [NotNullWhen(true)] out IWebSocketWriterHandl
             }
         }
 
+        /// <summary>
+        /// Checks whether the socket writer associated with the given key
+        /// is flagged as open. Actual connectivity is not checked.
+        /// </summary>
+        /// <remarks>
+        /// Implementation does not throw when key is not present in the underlying dictionary.
+        /// </remarks>
+        public bool IsSocketWriterOpen(TKey key)
+            => TryGetWriter(key, out var socketWriter) && !socketWriter.IsClosed;
```

Comment:
if the socket is closed we could also deregister it here, thoughts?

Reply:
The `WebSocketWriterRegistry` technically doesn't care about that. That's the job of the user of this type to call `Deregister` when the socket is closed (or if `Prune` is being called on a regular basis, which we don't at the moment). The one user of this _does do that_:

https://github.com/Azure/iotedge-lorawan-starterkit/blob/b50bf4fcb63e4ca4c0067a13ecf9a68ce67b9870/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/Processors/LnsProtocolMessageProcessor.cs#L163

I do feel this method should therefore be renamed and implemented such that it distinguishes between missing an entry, closed and open:

```c#
public bool? IsClosed(TKey key)
{
    lock (this.sockets)
        return this.sockets.TryGetValue(key, out var writer) ? writer.Object.IsClosed : null;
}
```

where the return value would be:

- `null` &rArr; unknown/unregistered
- `true` &rArr; closed
- `false` &rArr; open

Reply:
My comment might have been confusing, I didn't mean the `Deregister()` method here (as I understand this is for users of the registry). I rather meant `Prune()`. But we don't have to do that and eventually the closed sockets will be pruned.

About returning null here if the socket doesn't exist, do you have a use-case where that distinguish would be helpful in the current story/context? Because if not, I wouldn't do that yet.

Reply:
Let me know if we should hop on a call for this.

Reply:
Discussed in a call with Atif and our preferred way would be to create a private method that returns both objects and re-used from 2 places.

Reply:
As agreed on the call, let's either: 

1. Consolidate the duplication between `TryGetHandle` and `TryGetWriter`
2. Use `TryGetHandle`, make `IsSocketWriterOpen` private and expose `IsOpen` on `IWebSocketWriterHandle<>`.

Reply:
Done in e1bc1e1c9ba961a124531e8297a7476025a07a51